Dockerfile

```
@@ -101,4 +101,9 @@ WORKDIR ${RUNNER_PATH}
 
 # COMMAND
 ENV IN_DOCKER=1
-CMD ["cml-runner"]
+# Smart entrypoint understands commands like `bash` or `/bin/sh` but defaults to `cml`;
+# also works for GitLab CI/CD
+# https://gitlab.com/gitlab-org/gitlab-runner/-/blob/4c42e96/shells/bash.go#L18-37
+# https://gitlab.com/gitlab-org/gitlab-runner/-/blob/4c42e96/shells/bash.go#L288
+ENTRYPOINT ["/bin/bash", "-c", "basename \"$0\" | grep -qE '^(b?a|z|fi|t?c)?sh$' && exec \"$0\" \"$@\" || exec cml \"$0\" \"$@\""]
+CMD [""]
```

ok so back to earlier point, perhaps also this?

```suggestion
CMD ["runner"]
```

---

As per https://github.com/iterative/cml/pull/703#discussion_r700471745 and neighboring comments, I would prefer to keep `cml` neutral instead of using `runner` as the default commands. Users should be able to explore all the possible commands on the first interaction with the container.

---

Is there any behavioral difference between `CMD [""]` and `CMD ["--help"]` (?)

---

> Not enough non-option arguments: got 0, need at least 1 

this :point_up: